### ISSUE TITLE
Event detectors functions isolation from class structure

### ISSUE THREAD

**Issue by PMarmalyuk** [PMarmalyuk/EyeTrackingProject#2]
We need to make event detectors functions such as IVT, IDT, ANH, etc., independent of system's class structure. It is important to make it possible to use these functions by independent users/programmers.

**Comment by deslion** [comment]
Isolation is done. But need to test it in system. There are some problems, can't catch 'em.

**Comment by PMarmalyuk** [comment]
Could you please describe these problems arisen after the "isolation" modification? Maybe, let's discuss it in Skype?

**Comment by deslion** [comment]
I'll try to redownload the project again and find the problem.
I also have improved my little shiny-app - for all algorythms.

**Comment by deslion** [comment]
After starting the app and clicking "Accept all" button got the message: "Warning: Unhandled error in observer: argument is not interpretable as logical
observeEvent(input$acceptAllAssociations)"
Tried to check all the if-structures, but they are quite correct. This message was got in the version in the main repository. So I can't build subjects list.

**Comment by PMarmalyuk** [comment]
I have everything working... Try to locate the exact line which produces this warning. E.g. you can use print("Checkpoint 1"),... in different lines of observer to do this. Then you will figure out the chunk of code responsible for the warning.

Anyway, you can proceed to analyze data without setting associations!

**Comment by deslion** [comment]
I've found another moment - when I start the app, I get error messages like "could not find function". But as we can see all dependencies are set. I transfered lines "library(...)" to the ui.R and now it works. It's just some moment.

**Comment by deslion** [comment]
Located the problem - server.R lines between 647 and 650.

**Comment by PMarmalyuk** [comment]
Can't see any problems with this code... Can you check the values of the relevant variables coming to these lines? We need to figure out what is passed to class constructor (new(...)). Maybe the error is because something incorrect before these lines is producing incorrect values for the relevant variables.

**Comment by deslion** [comment]
So, afterall I found what's wrong. How did it happen:
1. Starting the app I've got message about "trimws".
2. ?trimws said: "This function is from memisc-package"
3. Installed memisc.
4. Nothing.
5. Googling "trimws" found this function in the base package v.3.2.2
6. Updated R.
7. Profit. And this was the main problem. Now it works erroring about different count of lines - but this is another story, work on it.
